spec/controllers/idv/verify_info_controller_spec.rb

```diff
@@ -349,6 +349,10 @@
               },
             ),
           )
+
+          event = @analytics.events["IdV: doc auth verify proofing results"].first
+          state_id = event[:proofing_results][:context][:stages][:state_id]
+          expect(state_id).to match(a_hash_including(state_id_type: 'drivers_license'))
```

Comment:
I couldn't find prior art for handling this situation...

I basically wanted:

```ruby
          expect(@analytics).to have_logged_event(
            'IdV: doc auth verify proofing results',
            hash_including(
              {
                proofing_results => context => stages => state_id => { state_id_type: 'drivers_license' }
              },
            ),
          )
```

except that's not valid code. I started nesting hash_including but it got _really_ unwieldy.

`hash_including` doesn't look deep in the hash, e.g., just doing `hash_including(state_id_type: 'drivers_license')` expects a top-level key named `state_id_type`; it doesn't dig down recursively until it finds a match.

The test above feels inelegant, except compared to every other option I tried.

Reply:
we have one double-nested hash-including here: https://github.com/18F/identity-idp/blob/0e9bba2636b9bf407e019003131870a5522ababd/spec/features/idv/analytics_spec.rb#L311

Reply:
It's almost like we need `have_logged_event` to support a pattern like:

```ruby
expect(@analytics).to have_logged_event(
    event_name, 
    hash_deeply_including(
        [:proofing_results, :context, :stages, :state_id],
        {state_id_type: 'drivers_license'},
    ),
)
```

Reply:
@lmgeorge Something like this came up in the `#rspec-league` channel when I asked about this, even with some example code. I'm reluctant to add it, though. I feel like it would normalize having deeply-nested hashes for important things, but I feel like it's an antipattern.

I think I agree more with the comment in that thread that something being clunky to test is often a smell that the code (or here, data structure) is too convoluted. That's not an easy fix in this PR, though.

Reply:
Agreed. This is one of those do you fix "all the things" or just try to make your current need work. Either way, adding a custom matcher or dramatically restructuring the analytics event is beyond the scope of this PR.